src/Sulu/Bundle/SecurityBundle/Resources/public/js/services/role-manager.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+define([
+    'services/husky/util',
+    'sulusecurity/models/role'
+], function(
+    Util,
+    Role
+) {
+
+    'use strict';
+
+    /**
+     * Saves a role with given data
+     *
+     * @param {Object} data The data to save
+     * @returns {Object} a promise
+     */
+    var save = function(data) {
+        var promise = $.Deferred(),
+            role = Role.findOrCreate({id: data.id});
+        role.set(data);
+
+        role.save(null, {
+            success: function(response) {
+                promise.resolve(response.toJSON());
+            }.bind(this),
+            error: function(context, jqXHR) {
+                promise.reject(jqXHR);
+            }.bind(this)
+        });
+
+        return promise;
+    },
+
+    /**
+     * Deletes a role for a given id
+     *
+     * @param roleId The id of the role to delete
+     * @returns {Object} a promise
+     */
+    remove = function(roleId) {
+        var role = Role.findOrCreate({id: roleId}),
+            promise = $.Deferred();
+
+        role.destroy({
+            success: function() {
+                promise.resolve();
+            }.bind(this),
+            error: function(context, jqXHR) {
+                promise.reject(getErrorStringForCode(jqXHR.responseJSON.code));
+            }.bind(this)
+        });
+
+        return promise;
+    },
+
+    /**
+     * For a given error code returns the corresponding
+     * error message.
+     *
+     * @param {Number} code The error code
+     * @returns {String} the corresponding error message
+     */
+    getErrorStringForCode = function(code) {
```

Comment:
Would you already do that here? This way the caller only has a translation code to react on the error. I would actually pass the entire `jqXHR` object, so that the caller has more options to react.